common/comms/proton_queue_adaptor.py

```diff
@@ -0,0 +1,106 @@
+"""Module for Proton specific queue adaptor functionality. """
+import uuid
+
+import tornado.ioloop
+import comms.queue_adaptor
+import proton.reactor
+import proton.handlers
+import utilities.integration_adaptors_logger as log
+
+logger = log.IntegrationAdaptorsLogger('PROTON_QUEUE')
+
+
+class MessageSendingError(RuntimeError):
+    """An error occurred whilst sending a message to the Message Queue"""
+    pass
+
+
+class EarlyDisconnectError(RuntimeError):
+    """The connection to the Message Queue ended before sending of the message had been done."""
+    pass
+
+
+class ProtonQueueAdaptor(comms.queue_adaptor.QueueAdaptor):
+    """Proton implementation of a queue adaptor."""
+
+    def __init__(self, **kwargs) -> None:
+        """
+        Constrcut a Proton implementation of a :class:`QueueAdaptor <comms.queue_adaptor.QueueAdaptor>`.
+        The kwargs provided should contain the following information:
+          * host: The host of the Message Queue to be interacted with.
+        :param kwargs: The key word arguments required for this constructor.
+        """
+        super(ProtonQueueAdaptor, self).__init__()
```

Comment:
[I don't think you need to pass any arguments](https://docs.python.org/3/library/functions.html#super):
```suggestion
        super().__init__()
```

Reply:
Done locally